src/modules/notifications/controllers/notifications.controller.ts

```diff
@@ -54,4 +79,120 @@ export class NotificationsController {
 		);
 		return { status: `updated ${amountUpdated} notifications`, total: amountUpdated };
 	}
+
+	@UseGuards(ApiKeyGuard)
+	@Post('check-new-notifications')
+	public async checkNewNotifications(
+		// eslint-disable-next-line @typescript-eslint/no-unused-vars
+		@Headers('apiKey') apiKey: string
+	): Promise<{
+		status: string;
+		notifications?: Partial<Record<NotificationType, number>>;
+		total: number;
+	}> {
+		const accessStartsNotifications = await this.sendAccessStartNotifications();
+		const accessAlmostEndedNotifications = await this.sendAccessAlmostEndedNotifications();
+		const accessEndedNotifications = await this.sendAccessEndNotifications();
+
+		const totalNotificationsSent =
+			accessStartsNotifications.length +
+			accessAlmostEndedNotifications.length +
+			accessEndedNotifications.length;
+		if (totalNotificationsSent > 0) {
+			return {
+				status: i18n.t('Notificaties verzonden'),
+				notifications: {
+					[NotificationType.ACCESS_PERIOD_READING_ROOM_STARTED]:
+						accessStartsNotifications.length,
+					[NotificationType.ACCESS_PERIOD_READING_ROOM_END_WARNING]:
+						accessAlmostEndedNotifications.length,
+					[NotificationType.ACCESS_PERIOD_READING_ROOM_ENDED]:
+						accessEndedNotifications.length,
+				},
+				total: totalNotificationsSent,
+			};
+		} else {
+			return {
+				status: i18n.t('No notifications had to be sent'),
+				total: 0,
+			};
+		}
+	}
+
+	/**
+	 * Check APPROVED visitRequests that have passed their start time, but not their end time and do not have a VISIT_REQUEST_ACCESS_STARTS notification
+	 * @private
+	 */
+	private async sendAccessStartNotifications(): Promise<Notification[]> {
+		const visits: Visit[] =
+			await this.visitService.getApprovedAndStartedVisitsWithoutNotification();
+		const notifications: GqlCreateOrUpdateNotification[] = visits.map(
+			(visit): GqlCreateOrUpdateNotification => ({
+				title: i18n.t('Je hebt nu toegang tot de leeszaal {{name}}', {
+					name: visit.spaceName,
+				}),
+				description: i18n.t('Je toegang vervalt terug op {{endDate}}', {
+					endDate: format(new Date(visit.endAt), 'dd/MM/yyyy HH:mm'),
+				}),
+				visit_id: visit.id,
```

Comment:
FE prefers snake_case instead of camelCase?

Reply:
yes, but this is an object that is passed into the this.notificationsService.create() function
the response of that function will use adaptNotification to correctly format the response for the frontend